atomic_reactor/util.py

```diff
@@ -626,3 +627,44 @@ def get_credentials(self, docker_registry):
         except KeyError:
             logger.warn('%s not found in .dockercfg', docker_registry)
             return {}
+
+
+def get_manifest_digests(image, insecure=False, dockercfg_path=None,
+                         versions=('v1', 'v2')):
+    """Return manifest digest for image.
+
+    :param image: ImageName, the remote image to inspect
+    :param insecure: bool, when True registry's cert is not verified
+    :param dockercfg_path: str, dirname of .dockercfg location
+    :param versions: tuple, which versions to fetch digest
+
+    :return: dict, versions mapped to their digest
+    """
+    auth = None
+    if dockercfg_path:
+        dockercfg = Dockercfg(dockercfg_path).get_credentials(image.registry)
+
+        username = dockercfg.get('username')
+        password = dockercfg.get('password')
+        if username and password:
+            auth = requests.auth.HTTPBasicAuth(username, password)
+
+    context = '/'.join([x for x in [image.namespace, image.repo] if x])
+    tag = image.tag or 'latest'
+    url = 'https://{}/v2/{}/manifests/{}'.format(image.registry, context, tag)
+
+    digests = {}
+    for version in versions:
+        headers = {
+            'Accept': 'application/vnd.docker.distribution.manifest.{}+json'.format(version)
+        }
+        response = requests.head(url, verify=not insecure, headers=headers,
+                                 auth=auth)
+        response.raise_for_status()
+
+        digests[version] = response.headers['Docker-Content-Digest']
```

Comment:
We don't know for sure which schema version we're getting. Maybe check the Content-Type header?